src/helpers.js

```diff
@@ -337,12 +337,27 @@ define(function (require) {
 
     return bufferSize;
   }
+ var safeBins =  p5.prototype.safeBins  = function(bins) {
+    let safeBins = 1024;
+    if(typeof(bins)==="string"){
+      console.log("the value of bins must be power of two and between 16 and 1024");
```

Comment:
should this be a `console.warn` or `console.error` ?

Reply:
a warning would be forgiving and error will remind user of its mistake ,
but here as even after the user enters wrong value we assign it a default one so it think , we are not breaking the operation so i think giving warning will be legitimate console.warn ()